logstash-core/lib/logstash/plugins/registry.rb

```diff
@@ -197,17 +198,24 @@ def lookup(type, plugin_name, &block)
     # plugin with the appropriate type.
     def legacy_lookup(type, plugin_name)
       begin
+        has_alias = @alias_registry.alias?(type.to_java, plugin_name)
```

Comment:
`has_alias` is used also to put in the `@registry` the loaded plugin with the alias name, so that the registry contains a plugin reference (the same) for the real name but also for the alias, https://github.com/elastic/logstash/pull/12796/files#diff-1395b4c417e22428a40924aa6e0196649a643a5a950b0b09e5d48408f97588bcR236